cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -115,9 +120,13 @@ bool MapBuilderBridge::HandleSubmapQuery(
   return true;
 }
 
-cartographer_ros_msgs::msg::SubmapList MapBuilderBridge::GetSubmapList() {
+cartographer_ros_msgs::msg::SubmapList MapBuilderBridge::GetSubmapList(
+    rclcpp::node::Node::SharedPtr& node) {
   cartographer_ros_msgs::msg::SubmapList submap_list;
-  submap_list.header.stamp = ::rclcpp::Time::now();
+  rclcpp::TimeSource ts(node);
```

Comment:
Sane here you should be able to reuse the time source. It has non-trivial initialization to subscribe to the clock topics etc.